schemas/2016-03-01/Microsoft.ServiceFabric.json

```diff
@@ -0,0 +1,640 @@
+{
+  "id": "http://schema.management.azure.com/schemas/2016-03-01/Microsoft.ServiceFabric.json#",
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "Microsoft.ServiceFabric",
+  "description": "Microsoft ServiceFabric Resource Types",
+  "resourceDefinitions": {
+    "clusters": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "enum": [
+            "Microsoft.ServiceFabric/clusters"
+          ]
+        },
+        "apiVersion": {
+          "enum": [
+            "2016-03-01"
+          ]
+        },
+        "properties": {
+          "properties": {
+            "VMImage": {
+              "type": "string",
+              "description": "Microsoft.ServiceFabric/clusters: The name of VM image VMSS has been configured with. Generic names such as Windows or Linux can be used"
+            },
+            "HttpApplicationGatewayCertificate": {
+              "oneOf": [
+                {
+                  "$ref": "#/definitions/CertificateDescription"
+                },
+                {
+                  "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+                }
+              ],
+              "description": "Microsoft.ServiceFabric/clusters: The server certificate used by http application gateway."
+            },
+            "AzureActiveDirectory": {
+              "oneOf": [
+                {
+                  "$ref": "#/definitions/AzureActiveDirectory"
+                },
+                {
+                  "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+                }
+              ],
+              "description": "Microsoft.ServiceFabric/clusters: The settings to enable AAD authentication on the cluster."
+            },
+            "ReliabilityLevel": {
+              "oneOf": [
+                {
+                  "enum": [
+                    "Invalid",
+                    "Bronze",
+                    "Silver",
+                    "Gold",
+                    "Platinum"
+                  ]
+                },
+                {
+                  "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+                }
+              ],
+              "description": "Microsoft.ServiceFabric/clusters: This level is used to set the number of replicas of the system services. Details: http://azure.microsoft.com/en-us/documentation/articles/service-fabric-cluster-capacity/#the-reliability-characteristics-of-the-cluster"
+            },
+            "NodeTypes": {
+              "oneOf": [
+                {
+                  "type": "array"
+                },
+                {
+                  "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+                }
+              ],
+              "items": {
```

Comment:
I mean, should we put the definition of "items" in the same object of "type": "array". Like this below, starting at line 66:
"oneOf": [
                {
                  "type": "array", 
                  "items": {
                  ......
                  }
                },
+                {
                  "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
                }
              ],